### PR TITLE
tweak wording around versioning

### LINKED ISSUE #826: Consolidate TEAL and AVM versioning
**Is your addition related to a problem? Please describe.**

Rather than continuing to disambiguate TEAL and AVM versions, the request is to consolidate to a single, AVM-based versioning scheme. The goal is to simplify the community's interactions with Algorand versioning.

**Describe the solution you'd like**

* The story requests updating docs references to reflect an AVM-based versioning scheme by removing  references to _TEAL version_.
* Where possible, use the language _program version_ to avoid specifying TEAL or AVM.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by barnjamin**
closes #826

### CODE REVIEW

File: docs/get-details/dapps/avm/teal/index.md [hunk 1]
```diff
@@ -12,8 +12,8 @@ Some of the opcodes in TEAL are only valid for a specific contract type. These a
...
+# Program versions
+Currently, Algorand supports versions 1 through 6 of TEAL. When writing contracts with Program version 2 or higher, make sure to add `#pragma version #` where # should be replaced by the specific number, as the first line of the program. If this line does not exist, the protocol will treat the contract as a version 1 contract. If upgrading a contract to version 2 or higher, it is important to verify you are checking the `RekeyTo` property of all transactions that are attached to the contract. See the [TEAL guidelines](guidelines) for more suggestions to help prevent a contract from being compromised.
```
Comment (michaeldiamant) [inline docs/get-details/dapps/avm/teal/index.md:16]: @barnjamin Is it possible to link elsewhere to show current AVM versions?

> Algorand supports versions 1 through 6 of TEAL

* I imagine the max version is liable to go stale.  
* I can also imagine you deliberately wish to provide an inline reference.  If that's the case, I leave the following _suggestion_.

```suggestion
Currently, Algorand supports program versions 1 through 6. When writing contracts with Program version 2 or higher, make sure to add `#pragma version #` where # should be replaced by the specific number, as the first line of the program. If this line does not exist, the protocol will treat the contract as a version 1 contract. If upgrading a contract to version 2 or higher, it is important to verify you are checking the `RekeyTo` property of all transactions that are attached to the contract. See the [TEAL guidelines](guidelines) for more suggestions to help prevent a contract from being compromised. 
```

Comment (michaeldiamant) [inline docs/get-details/dapps/avm/teal/index.md:16]: @barnjamin Optional - I don't feel _Program_ must be capitalized.  Feel welcomed to resolve as is.
```suggestion
Currently, Algorand supports versions 1 through 6 of TEAL. When writing contracts with program version 2 or higher, make sure to add `#pragma version #` where # should be replaced by the specific number, as the first line of the program. If this line does not exist, the protocol will treat the contract as a version 1 contract. If upgrading a contract to version 2 or higher, it is important to verify you are checking the `RekeyTo` property of all transactions that are attached to the contract. See the [TEAL guidelines](guidelines) for more suggestions to help prevent a contract from being compromised. 
```

File: docs/get-details/dapps/avm/teal/index.md [hunk 2]
```diff
@@ -12,8 +12,8 @@ Some of the opcodes in TEAL are only valid for a specific contract type. These a
 !!! warning
     When writing smart contracts, make sure to follow [TEAL guidelines](guidelines). This is very important in order to prevent smart contracts from being compromised.
```
Comment (michaeldiamant) [inline docs/get-details/dapps/avm/teal/index.md:13]: @barnjamin I found several additional references to consider.  Can you review https://github.com/algorand/docs/pull/878 to see if you agree with pulling it in?

File: docs/get-details/dapps/avm/teal/index.md [hunk 3]
```diff
@@ -12,8 +12,8 @@ Some of the opcodes in TEAL are only valid for a specific contract type. These a
 !!! warning
```
[the remark this answers is not in the document]
Comment (barnjamin) [reply docs/get-details/dapps/avm/teal/index.md:12]: I expect to do a larger re-org of the docs to flatten the structure a bit, will make note of this change for when that happens

File: docs/get-details/dapps/avm/teal/index.md (hunk 1 above)
Comment (barnjamin) [reply docs/get-details/dapps/avm/teal/index.md]: I agree we need to find a way to automatically keep this up to date. Either with some script or a link. Will make note and consider it outside this pr

**Review by michaeldiamant — APPROVED** [review]
@barnjamin Thanks for the quick pass here to prep for AVM 7 release!

Leaving approval with the understanding that a broader look at AVM vs TEAL _docs_ structure will happen out-of-band to the PR.

**Pull request merged**